tests/EndToEnd/PsalmRunnerTrait.php

```diff
@@ -42,7 +42,7 @@ private function runPsalm(
             $process->mustRun();
         } else {
             $process->run();
-            $this->assertGreaterThan(0, $process->getExitCode());
+            $this->assertEquals(2, $process->getExitCode());
```

Comment:
As Psalm now always return exit code 2 when it detects issues, we expect that when `$shouldFail === true`, and let unexpected exceptions be reported as test failures.